dht_test.go

```diff
@@ -32,7 +32,11 @@ func ExampleDHT() {
 		fmt.Println(err)
 		return
 	}
-	go d.Run()
+	err = d.Start()
```

Comment:
minor: could you inline if err checking here and in the places you're handling errors below?

if err = d.Start(); err != nil {